### ISSUE TITLE
0021_set_reviewer_data.py migrations fail when there is no suggestion.reviewer

### ISSUE THREAD

**Issue by dwaynebailey** [translate/pootle#3483]
The following Traceback occurs in ``pootle_store/migrations/0021_set_reviewer_data.py`` when migrating http://pootle.locamotion.org to ``stable/2.6.0``.  This is our oldest DB so I'd expect some fallout as it likely carries across data that doesn't meet our migration assumptions.

It seems that we have suggestions with no submitter and that causes a failure.

```sql
2015-01-15 17:09:57,979 DEBUG (0.048) INSERT INTO `auth_permission` (`name`, `content_type_id`, `codename`) VALUES ('Can add score log', 44, 'add_scorelog'), ('Can change score log', 44, 'change_scorelog'), ('Can delete score log', 44, 'delete_scorelog'); args=(u'Can add score log', 44, u'add_scorelog', u'Can change score log', 44, u'change_scorelog', u'Can delete score log', 44, u'delete_scorelog')
2015-01-15 17:09:58,127 DEBUG (0.056) SELECT `django_content_type`.`id`, `django_content_type`.`name`, `django_content_type`.`app_label`, `django_content_type`.`model` FROM `django_content_type` WHERE `django_content_type`.`app_label` = 'pootle_statistics'  ORDER BY `django_content_type`.`name` ASC; args=(u'pootle_statistics',)
```

```pytb
Traceback (most recent call last):
  File "./manage.py", line 32, in <module>
    execute_from_command_line()
  File "/home/vagrant/.virtualenvs/pootle/local/lib/python2.7/site-packages/django/core/management/__init__.py", line 399, in execute_from_command_line
    utility.execute()
  File "/home/vagrant/.virtualenvs/pootle/local/lib/python2.7/site-packages/django/core/management/__init__.py", line 392, in execute
    self.fetch_command(subcommand).run_from_argv(self.argv)
  File "/home/vagrant/.virtualenvs/pootle/local/lib/python2.7/site-packages/django/core/management/base.py", line 242, in run_from_argv
    self.execute(*args, **options.__dict__)
  File "/home/vagrant/.virtualenvs/pootle/local/lib/python2.7/site-packages/django/core/management/base.py", line 285, in execute
    output = self.handle(*args, **options)
  File "/home/vagrant/.virtualenvs/pootle/local/lib/python2.7/site-packages/django/core/management/base.py", line 415, in handle
    return self.handle_noargs(**options)
  File "/vagrant/pootle/apps/pootle_app/management/commands/setup.py", line 85, in handle_noargs
    call_command('migrate', interactive=False)
  File "/home/vagrant/.virtualenvs/pootle/local/lib/python2.7/site-packages/django/core/management/__init__.py", line 159, in call_command
    return klass.execute(*args, **defaults)
  File "/home/vagrant/.virtualenvs/pootle/local/lib/python2.7/site-packages/django/core/management/base.py", line 285, in execute
    output = self.handle(*args, **options)
  File "/home/vagrant/.virtualenvs/pootle/local/lib/python2.7/site-packages/south/management/commands/migrate.py", line 111, in handle
    ignore_ghosts = ignore_ghosts,
  File "/home/vagrant/.virtualenvs/pootle/local/lib/python2.7/site-packages/south/migration/__init__.py", line 220, in migrate_app
    success = migrator.migrate_many(target, workplan, database)
  File "/home/vagrant/.virtualenvs/pootle/local/lib/python2.7/site-packages/south/migration/migrators.py", line 256, in migrate_many
    result = migrator.__class__.migrate_many(migrator, target, migrations, database)
  File "/home/vagrant/.virtualenvs/pootle/local/lib/python2.7/site-packages/south/migration/migrators.py", line 331, in migrate_many
    result = self.migrate(migration, database)
  File "/home/vagrant/.virtualenvs/pootle/local/lib/python2.7/site-packages/south/migration/migrators.py", line 133, in migrate
    result = self.run(migration, database)
  File "/home/vagrant/.virtualenvs/pootle/local/lib/python2.7/site-packages/south/migration/migrators.py", line 114, in run
    return self.run_migration(migration, database)
  File "/home/vagrant/.virtualenvs/pootle/local/lib/python2.7/site-packages/south/migration/migrators.py", line 84, in run_migration
    migration_function()
  File "/home/vagrant/.virtualenvs/pootle/local/lib/python2.7/site-packages/south/migration/migrators.py", line 60, in <lambda>
    return (lambda: direction(orm))
  File "/vagrant/pootle/apps/pootle_store/migrations/0021_set_reviewer_data.py", line 40, in forwards
    s.submitter = s.suggestion.reviewer
AttributeError: 'NoneType' object has no attribute 'reviewer'
```

**Comment by jleclanche** [comment]
These are Submissions without a suggestion item. Is that to be expected? The field is marked as null=True in `pootle_statistics/models.py` but it seems weird, doesn't it?

**Comment by unho** [comment]
@jleclanche Submissions can be for a translation, so they might not have a suggestion.

**Comment by jleclanche** [comment]
Fixed

**Comment by dwaynebailey** [comment]
@jleclanche fixed in 6081bbe8fc0cdf433c320c21a3e5b18ec4bfa62c

Lets use the amazingness of github to save puppies dying when some poor sod has to look at this sometime in the future.

**Comment by jleclanche** [comment]
You're right, I'm lazy and very sorry ;)
